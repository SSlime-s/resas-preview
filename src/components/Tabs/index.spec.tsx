import { render, fireEvent } from "@testing-library/react";

import { Tabs } from ".";

function setup() {
	const onChange = vi.fn();
	const result = render(
		<Tabs
			idPrefix="tablist"
			panelIdPrefix="tabpanel"
			options={[
				{ label: "Foo", value: "foo" },
				{ label: "Bar", value: "bar" },
			]}
			selected="foo"
			onChange={onChange}
		/>
	);

	return { ...result, mock: { onChange } };
}

describe("Components/Tabs", () => {
	it("スナップショット", () => {
		const { container } = setup();

		expect(container).toMatchSnapshot();
	});

	it("クリックでタブを切り替えられる", () => {
		const {
			getByRole,
			mock: { onChange },
		} = setup();

		const tab = getByRole("tab", { name: "Bar" });
		tab.click();

		expect(onChange).toHaveBeenCalledWith("bar");
	});

	it("キーボード操作でタブを切り替えられる", () => {
		const {
			getByRole,
			mock: { onChange },
		} = setup();

		const tab = getByRole("tab", { name: "Foo" });
		tab.focus();
		fireEvent.keyDown(tab, { key: "ArrowRight" });

		expect(onChange).toHaveBeenCalledWith("bar");
	});
});
