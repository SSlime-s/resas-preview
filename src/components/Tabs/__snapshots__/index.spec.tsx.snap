// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components/Tabs > スナップショット 1`] = `
<div>
  <div
    class="🐻-1295175268"
    role="tablist"
  >
    <button
      aria-controls="tabpanel-foo"
      aria-selected="true"
      class="🐻-2374718740"
      id="tablist-foo"
      role="tab"
      tabindex="0"
    >
      Foo
    </button>
    <button
      aria-controls="tabpanel-bar"
      aria-selected="false"
      class="🐻-2374718740"
      id="tablist-bar"
      role="tab"
      tabindex="-1"
    >
      Bar
    </button>
  </div>
</div>
`;
